features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/members/RoomMemberListStateProvider.kt

```diff
@@ -17,27 +17,88 @@
+                        joined = persistentListOf(anAlice(), aBob()),
+                    )
+                )
+            ),
+            aRoomMemberListState(roomMembers = Async.Loading()),
+            aRoomMemberListState().copy(isSearchActive = false),
+            aRoomMemberListState().copy(isSearchActive = true),
+            aRoomMemberListState().copy(isSearchActive = true, searchQuery = "someone"),
+            aRoomMemberListState().copy(
+                isSearchActive = true,
+                searchQuery = "@someone:matrix.org",
+                searchResults = RoomMemberSearchResultState.Results(aRoomMemberList().toImmutableList()),
+            ),
+            aRoomMemberListState().copy(
+                isSearchActive = true,
+                searchQuery = "something-with-no-results",
+                searchResults = RoomMemberSearchResultState.NoResults
+            ),
         )
 }
 
 internal fun aRoomMemberListState(
-    searchResults: UserSearchResultState = UserSearchResultState.NotSearching,
-    allUsers: Async<ImmutableList<MatrixUser>> = Async.Uninitialized,
-) =
-    RoomMemberListState(
-        userListState = aUserListState().copy(searchResults = searchResults),
-        allUsers = allUsers,
-    )
+    roomMembers: Async<RoomMembers> = Async.Uninitialized,
+    searchResults: RoomMemberSearchResultState = RoomMemberSearchResultState.NotSearching,
+) = RoomMemberListState(
+    roomMembers = roomMembers,
+    searchQuery = "",
+    searchResults = searchResults,
+    isSearchActive = false,
+    eventSink = {}
+)
+
+fun aRoomMember(
+    userId: UserId = UserId("@alice:server.org"),
+    displayName: String? = null,
+    avatarUrl: String? = null,
+    membership: RoomMembershipState = RoomMembershipState.JOIN,
+    isNameAmbiguous: Boolean = false,
+    powerLevel: Long = 0L,
+    normalizedPowerLevel: Long = 0L,
+    isIgnored: Boolean = false,
+) = RoomMember(
+    userId = userId,
+    displayName = displayName,
+    avatarUrl = avatarUrl,
+    membership = membership,
+    isNameAmbiguous = isNameAmbiguous,
+    powerLevel = powerLevel,
+    normalizedPowerLevel = normalizedPowerLevel,
+    isIgnored = isIgnored,
+)
+
+fun aRoomMemberList() = listOf(
+    anAlice(),
+    aBob(),
+    aRoomMember(UserId("@carol:server.org"), "Carol"),
+    aRoomMember(UserId("@david:server.org"), "David"),
+    aRoomMember(UserId("@eve:server.org"), "Eve"),
+    aRoomMember(UserId("@justin:server.org"), "Justin"),
+    aRoomMember(UserId("@mallory:server.org"), "Mallory"),
+    aRoomMember(UserId("@susie:server.org"), "Susie"),
+    aVictor(),
+    aWalter(),
+)
+
+fun anAlice() = aRoomMember(UserId("@alice:server.org"), "Alice")
+fun aBob() = aRoomMember(UserId("@bob:server.org"), "Bob")
+
+fun aVictor() = aRoomMember(UserId("@victor:server.org"), "Victor", membership = RoomMembershipState.INVITE)
+
+fun aWalter() = aRoomMember(UserId("@walter:server.org"), "Walter", membership = RoomMembershipState.INVITE)
```

Comment:
I am not sure if we want to expose them to the rest of the module/app, maybe we can have these methods private?

Reply:
Some of them are in use by other tests

Reply:
These functions are in the impl module which is not supposed to be imported by the other modules, so this is definitely not a problem